File: C++/LabWork#1/ExpressionOperations.cpp
#include "ExpressionOperations.h"
#include <algorithm>

using namespace std;

ExpressionOperations::ExpressionOperations(string input) {
  text = input;
  
  transform(text.begin(), text.end(), text.begin(), ::toupper);
}

string ExpressionOperations::GetText() const {
  return text;
}

int ExpressionOperations::CalculateExpressionLength() const{
  return text.length();
}

string ExpressionOperations::ReplaceChars(char oldChar, char newChar) {
  string result = text;
  replace(result.begin(), result.end(), oldChar, newChar);
  return result;
}

File: C#/LabWork#1/ExpressionOperations.cs
using System;

namespace LabWork_1
{
  public class ExpressionOperations
  {
    private string text;

    public ExpressionOperations(string input)
    {
      text = input.ToUpper();
    }
    public string GetText()
    {
      return text;
    }
    public int CalculateExpressionLength()
    {
      return text.Length;
    }
    public string ReplaceChars(char oldChar, char newChar) {

      return text.Replace(oldChar, newChar);

    }
  }
}
